pyinteraph/centrality_analysis.py

```diff
@@ -0,0 +1,604 @@
+
+    centrality_val = nxc.group_closeness_centrality(G = G,
+                                                    S = kwargs['node_list'],
+                                                    weight = kwargs['weight_name'])
+    centrality_dict = get_dict_with_group_val(G, kwargs['node_list'], centrality_val)
+    return centrality_dict
+
+def get_edge_betweenness_cent(G, **kwargs):
+    """Returns a dictionary of edge betweenness centrality values."""
+    
+    centrality_dict = nxc.edge_betweenness_centrality(G = G,
+                                                      normalized = kwargs['norm'],
+                                                      weight = kwargs['weight_name'])
+    return centrality_dict
+
+def get_centrality_dict(cent_list, function_map, graph, **kwargs):
+    """Returns two dictionaries. For the first dictionary, the key is the 
+    name of a node centrality measure and the value is a dictionary of 
+    centrality values for each node. (Also includes group centralities)
+    e.g. {degree: {A: 0.1, B:0.7, ...}, betweenness: {...}, ...}
+    For the second dictionary, the key is the name of an edge centrality
+    measure and the value is a dictionary of centrality values for each
+    edge, similar to the first dictionary.
+    """
+
+    node_dict = {}
+    edge_dict = {}
+    for name in cent_list:
+        # Get dictionary using the function map
+        cent_dict = function_map[name](G = graph, **kwargs)
+        # Add edge centralities to edge dict
+        if "edge" in name:
+            edge_dict[name] = cent_dict
+        # Add node centralities to node dict
+        else:
+            node_dict[name] = cent_dict
+    return node_dict, edge_dict
+
+def write_table(fname, centrality_dict, identifiers, sort_by):
+    """Takes in a dictionary of dictionaries and saves a file where each 
+    row consists of a node and its corresponding centrality values.
+    """
+
+    # Remove any file extensions
+    fname = os.path.splitext(fname)[0]
+
+    with open(f"{fname}.txt", "w") as f:
+        # Add first line (header)
+        line = f"node"
+        for key in centrality_dict.keys():
+            # Add name of each centrality
+            line += f"\t{key}"
+        line += "\n"
+        f.write(line)
+        # Choose order of nodes
+        if sort_by == "node":
+            sorted_nodes = identifiers
+        else:
+            sorted_dict = sorted(centrality_dict[sort_by].items(), 
+                                key = lambda tup: tup[1],
+                                reverse = True)
+            sorted_nodes = [n for (n, v) in sorted_dict]
+        # for node in identifiers, add corresponding row:
+        for node in sorted_nodes:
+            line = f"{node}"
+            for c_dict in centrality_dict.values():
+                # Add each centrality value
+                line += f"\t{c_dict[node]}"
+            line += "\n"
+            f.write(line)
+
+def replace_bfac_column(pdb, vals, pdb_out):
```

Comment:
You can import this function from `graph_analysis` if it is already there. The same is true for graph building. It is better not to have duplicate code, and shared functions will eventually go in a separate script anyway.

Reply:
The `build_graph` function is slightly different from the one in `graph_analysis`  but exactly the same as the one in `path_analysis`. So should I change the `build_graph` function in `graph_analysis` or wait until `path_analysis` is merged and import it from there?

Reply:
Sorry, you are right! I think we can wait for `path_analysis` to be merged and then import it from there :) let's leave this comment open then